trie/zk_trie.go

```diff
@@ -180,7 +180,7 @@ func (t *ZkTrie) Prove(key []byte, fromLevel uint, proofDb ethdb.KeyValueWriter)
 			return err
 		}
 
-		if n.Type == zktrie.NodeTypeLeaf {
+		if n.Type == zktrie.NodeTypeLeaf_New {
```

Comment:
It's quite error prone to leave both `NodeTypeLeaf` and `NodeTypeLeaf_New` in `zktrie`. We can leave it as it is now but it would have been better to remove `NodeTypeLeaf` so that it cannot be used accidentally.

Reply:
This issue has been traced in zktrie now: https://github.com/scroll-tech/zktrie/issues/15

What about to update the defination along with the next upgrading of zktrie?

Reply:
Yes, let's update this in l2geth later.